### PR TITLE
Aviral

### PR DESCRIPTION
**Pull request by AviralJ58**
Product route working correctly.
GET (all),GET (by id), POST, PATCH, DELETE routes created.

### CODE REVIEW

File: Files/product/routes.py [hunk 1]
```diff
@@ -1,73 +1,24 @@
...
+from .utils import get_all_products, get_product_by_id, add_product, update_product, remove_product
 
 product = Blueprint('product', __name__, url_prefix='/product')
 
-from .utils import add_product_args, resource_fields
-
 @product.get('/')
-# @marshal_with(resource_fields)
 def get_products():
-    result=Product.query.all()
```
Comment (sidd6p) [inline Files/product/routes.py:14]: > do in this manner, this 
```py
    result=get_all_products()
    if result is None:
       return {"message": "There are 0 products"}, status_code
    return jsonify(result), status_code
```

File: Files/product/routes.py [hunk 2]
```diff
@@ -1,73 +1,24 @@
...
-        product=Product(name=request.json['name'],description=request.json['description'],price=request.json['price'],
-            image=request.json['image'],discount=request.json['discount'],effective_price=float(request.json['price'])-(float(request.json['discount'])*float(request.json['price'])/100),
-            qty_left=request.json['qty_left'],category=request.json['category'],related_products=request.json['related_products'])
-        db.session.add(product)
-        db.session.commit()
-        return {"message": "Dones"}, 201
-
-    return {"message": "Request must be JSON"}, 415
+def post_product():
+    return add_product()
```
Comment (sidd6p) [inline Files/product/routes.py:16]: > extract the data from the request and then send it to the add_product(), no need to send the request object itself

**Review by sidd6p — CHANGES_REQUESTED** [review]
make changes

File: Files/product/routes.py (hunk 2 above)
Comment (AviralJ58) [reply Files/product/routes.py]: Fixed in the latest commit.

File: Files/product/routes.py (hunk 1 above)
Comment (AviralJ58) [reply Files/product/routes.py:14]: Fixed in the latest commit.

**Pull request merged**